### ISSUE TITLE
Do you have Python code

### ISSUE THREAD

**Issue by kim-yhow** [harvardnlp/im2markup#26]
To be honest, I'm not very familiar with Lua code, so I wish you could have a python implementation. It’s very important for me. Thank you very much

**Comment by da03** [comment]
yeah we have it implemented in python as well: http://opennmt.net/OpenNMT-py/im2text.html

**Comment by kim-yhow** [comment]
> yeah we have it implemented in python as well: http://opennmt.net/OpenNMT-py/im2text.html

Thank you for your code. And I download the trained model and dataset which the link mentioned. And evaluate it with the test dataset . However the results can not  reach the scores mentioned in the paper.
![results](https://user-images.githubusercontent.com/25976152/69692273-6f21fe80-110c-11ea-9b17-10d299c4dabc.png)
tips: the evaluation code is from [https://github.com/harvardnlp/im2markup/tree/master/scripts/evaluation](url)
![}2WJ0K2V1RLQU)7T4 _0VYS](https://user-images.githubusercontent.com/25976152/69692357-bc9e6b80-110c-11ea-980a-a080709aca99.png)

**Comment by da03** [comment]
Hmm interesting. There are some differences between the python code and the lua code, e.g., the batching is different, and the python code is treating images as three channels R,G,B, whereas the lua code converts images to greyscale first which decreases the number of parameters. For reproducing the results in the paper, I'd suggest using the lua code which I'm pretty sure can get the reported accuracy.

**Comment by kim-yhow** [comment]
Thanks, I will try to use the lua code. And do you have the model which you have trained by lua code?

**Comment by da03** [comment]
Yes see this section: https://github.com/harvardnlp/im2markup#test

**Comment by kim-yhow** [comment]
Oh, yeah. I found it, thank you very much！ And by the way, I wonder if the trained model in **python** is the best one. In other word, if I train it for more time,  will  it perform better?

**Comment by da03** [comment]
I'm not sure since I trained it years ago... But please let me know if you can get a pytorch model matching the performance of the luatorch one. (I think it might be possible to convert the luatorch model to pytorch if we manually convert all weights)

**Comment by kim-yhow** [comment]
Okey, Thank you!

**Comment by kim-yhow** [comment]
Hi, when i run the lua code, I found the following problem. 
1. Why the accuracy is nan? The accuracy of the first 500 is 0.93， however , the following of accuracy is nan  
![1](https://user-images.githubusercontent.com/25976152/69729007-8ab6f480-1160-11ea-9893-d8ea4d706897.png)
2. What is the meaning of the following images. 
![2](https://user-images.githubusercontent.com/25976152/69729012-8c80b800-1160-11ea-8c88-58b8fb73ef02.png)
Thank you very much!

**Comment by kim-yhow** [comment]
Thank you very much, and I reproduce the results successfully with lua code.

**Comment by da03** [comment]
closing now, please reopen if there are other problems.

**Comment by PhenomenalOnee** [comment]
This http://opennmt.net/OpenNMT-py/im2text.html
link is not working now
Do you have the python code to run the pre trained 
models

**Comment by StephenKyung** [comment]
> This http://opennmt.net/OpenNMT-py/im2text.html
> link is not working now
> Do you have the python code to run the pre trained
> models

link has chnaged! check this :
https://opennmt.net/OpenNMT-py/legacy/im2text.html
